### PR TITLE
fix: disable battery segment when no batteries

### PR DESCRIPTION
**Pull request by lnu**
### Prerequisites

- [x] I have read and understand the `CONTRIBUTING` guide
- [x] The commit message follows the [conventional commits][cc] guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

I tested the latest version on a desktop without batteries(opensuse tumbleweed, zsh latest, omp latest) and it doesn't return any error but an empty batteries list. I did a quick fix for it but I'm not sure it covers all scenarios.

[cc]: https://www.conventionalcommits.org/en/v1.0.0/#summary

### CODE REVIEW

File: src/segment_battery.go [hunk 1]
```diff
@@ -34,7 +34,7 @@ const (
 
 func (b *batt) enabled() bool {
 	batteries, err := b.env.getBatteryInfo()
-	if !b.enabledWhileError(err) {
+	if !b.enabledWhileError(err) || len(batteries) == 0 {
```
Comment (JanDeDobbeleer) [inline src/segment_battery.go]: I did mis this when refactoring. Should be good (although the tests probably don't cover the use-case)

Comment (JanDeDobbeleer) [reply src/segment_battery.go]: @lnu can you add a tests that ensures this is validated as well?

**Review by JanDeDobbeleer — CHANGES_REQUESTED** [review]

**Review by JanDeDobbeleer — APPROVED** [review]

**Comment by JanDeDobbeleer** [comment]
@lnu can you rebase?

**Comment by lnu** [comment]
Will be my first job tomorrow 🙂

**Pull request merged**